src/software/ai/motion_constraint/motion_constraint_set_builder.cpp

```diff
@@ -0,0 +1,77 @@
+#include "motion_constraint_set_builder.h"
+
+std::set<MotionConstraint> MotionConstraintSetBuilder::buildMotionConstraints(
+    const GameState &game_state, Tactic &tactic)
+{
+    current_allowed_constraints.clear();
+    std::set<MotionConstraint> current_motion_constraints =
+        buildMotionConstraintsFromGameState(game_state);
+    try
+    {
+        // updates current_allowed_constraints
+        motion_constraint_visitor.set_current_allowed_constraint(
+            current_allowed_constraints);
+        tactic.accept(motion_constraint_visitor);
+    }
+    catch (std::invalid_argument)
+    {
+        // Tactic didn't implement accept so don't add any more motion constraints
```

Comment:
Not sure what the scope of the PR is, but maybe add at least some logging here so we know that it happened

Reply:
I think both of your points are good but I believe it is out of scope for this PR.